things/clients/api/transport.go

```diff
@@ -18,95 +18,96 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"go.opentelemetry.io/contrib/instrumentation/github.com/go-kit/kit/otelkit"
 )
 
 // MakeHandler returns a HTTP handler for API endpoints.
-func MakeHandler(svc clients.Service, mux *bone.Mux, logger mflog.Logger) http.Handler {
+func MakeHandler(csvc clients.Service, psvc policies.Service, mux *bone.Mux, logger mflog.Logger) http.Handler {
 	opts := []kithttp.ServerOption{
 		kithttp.ServerErrorEncoder(apiutil.LoggingErrorEncoder(logger, api.EncodeError)),
 	}
 
 	mux.Post("/things", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("create_thing"))(createClientEndpoint(svc)),
+		otelkit.EndpointMiddleware(otelkit.WithOperation("create_thing"))(createClientEndpoint(csvc)),
 		decodeCreateClientReq,
 		api.EncodeResponse,
 		opts...,
 	))
 
 	mux.Post("/things/bulk", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("create_things"))(createClientsEndpoint(svc)),
+		otelkit.EndpointMiddleware(otelkit.WithOperation("create_things"))(createClientsEndpoint(csvc)),
 		decodeCreateClientsReq,
 		api.EncodeResponse,
 		opts...,
 	))
 
 	mux.Get("/things/:thingID", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("view_thing"))(viewClientEndpoint(svc)),
+		otelkit.EndpointMiddleware(otelkit.WithOperation("view_thing"))(viewClientEndpoint(csvc)),
 		decodeViewClient,
 		api.EncodeResponse,
 		opts...,
 	))
 
 	mux.Get("/things", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("list_things"))(listClientsEndpoint(svc)),
+		otelkit.EndpointMiddleware(otelkit.WithOperation("list_things"))(listClientsEndpoint(csvc)),
 		decodeListClients,
 		api.EncodeResponse,
 		opts...,
 	))
 
 	mux.Get("/channels/:thingID/things", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("list_things_by_channel"))(listMembersEndpoint(svc)),
+		otelkit.EndpointMiddleware(otelkit.WithOperation("list_things_by_channel"))(listMembersEndpoint(csvc)),
 		decodeListMembersRequest,
 		api.EncodeResponse,
 		opts...,
 	))
 
 	mux.Patch("/things/:thingID", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("update_thing_name_and_metadata"))(updateClientEndpoint(svc)),
+		otelkit.EndpointMiddleware(otelkit.WithOperation("update_thing_name_and_metadata"))(updateClientEndpoint(csvc)),
 		decodeUpdateClient,
 		api.EncodeResponse,
 		opts...,
 	))
 
 	mux.Patch("/things/:thingID/tags", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("update_thing_tags"))(updateClientTagsEndpoint(svc)),
+		otelkit.EndpointMiddleware(otelkit.WithOperation("update_thing_tags"))(updateClientTagsEndpoint(csvc)),
 		decodeUpdateClientTags,
 		api.EncodeResponse,
 		opts...,
 	))
 
-	mux.Post("/things/:thingID/share", kithttp.NewServer(
-		otelkit.EndpointMiddleware(otelkit.WithOperation("share_thing"))(shareClientEndpoint(svc)),
+	mux.Post("/things/share", kithttp.NewServer(
```

Comment:
No need to expose this API at all; since it's already exposed at Policies. UI/CLI will take care of using the correct endpoint.